### PR TITLE
Hyperopt->Raytune for Hyperparameter Optimization and Different Objective Function

### PR DESCRIPTION
**Pull request by stompsjo**
A refactor of hyperparameter optimization codes for ML models to use `ray[tune]` instead of `hyperopt`. See the [raytune](https://docs.ray.io/en/latest/tune/index.html) docs for more information. The main feature additions, and the motivation for the chance, are better compatibility for parallel computation and other options for hyperparameter optimization algorithms. For example, `ray[tune]` has a wrapper function for the same `hyperopt` algorithm used before, but also has other algorithms like Bayesian Optimization HyperBand which could be used in the future. Raytune also allows a user to specify the number of trials to run simultaneously, which greatly increases the speed or capabilities for testing different hyperparameters on powerful computers.

This should be reviewed after the original ML/SSML models have been reviewed and merged, since this is a feature change from the original exploratory work and is a refactor of existing code. **Note:** hyperparameter optimization now _maximizes_ a score rather than _minimizing_ a loss as before (even though both are reported to raytune. The new objective function/score is:

```
score = balanced_accuracy + alpha*recall + beta*precision
```

This allows a user to reward increases in precision or recall asymmetrically. That is, if a user cares more about reducing false positives over false negatives, or vice versa, they can change alpha/beta in optimization. This may take longer to optimization, since the state space is increasing, but provides more flexibility in model evaluation than purely based on balanced accuracy.

To address:
- [ ] Complete code coverage with unit tests.
- [ ] Consider how to make unit tests faster with lightweight models.
- [ ] Consider making it possible to use more than just `HyperOptSearch` for optimization.
- [ ] Consider the ability to use a [Scheduler](https://docs.ray.io/en/latest/tune/api_docs/schedulers.html) or checkpointing during training loop(s) with `tune.session.report()`.

### CODE REVIEW

**Comment by stompsjo** [comment]
I will push this branch `stompsjo/RadClass:raytune` to `cnerg:RadClass:raytune` for posterity, but I reverted to using hyperopt when I did my dissertation research, meaning this PR is moot. In the time since I created this PR, raytune stopped playing nicely with the rest of my code, meaning I never managed to get parallel hyperparameter optimization trials working. Instead, I used `hyperopt` and initiated several jobs on UW-Madison's CHTC.
